python/hail/docs/tutorials/introduction-to-the-expression-language.ipynb

```diff
@@ -0,0 +1,1465 @@
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Perhaps we just want the set of unique genes:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "hc.eval_expr_typed('va.transcripts.map(t => t.gene).toSet()')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can find the canonical transcript with `find`, which returns the first element where the predicate is true:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "hc.eval_expr_typed('va.transcripts.find(t => t.canonical)')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "However, `find` returns `None` if there isn't an element where the predicate is true:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "hc.eval_expr_typed('va.transcripts.find(t => t.gene == \"GENE5\")')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, we'll pull out all transcripts marked \"MIS\" (missense):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "hc.eval_expr_typed('va.transcripts.filter(t => t.consequence == \"MIS\")')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here's a bit of a complex motif - we can sort the transcripts by an arbitrary function. Here we'll sort so that `\"LOF\"` comes before `\"MIS\"`, and `\"MIS\"` comes before `\"SYN\"`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "hc.eval_expr_typed('''va.transcripts.sortBy(t => \n",
+    "                        if (t.consequence == \"LOF\") 1 \n",
+    "                        else if (t.consequence == \"MIS\") 2 \n",
+    "                        else 3)''')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If we are interested in pulling out the worst-consequence transcript, we can use this sorting motif and then take the first element:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "hc.eval_expr_typed('''va.transcripts.sortBy(t => \n",
+    "                        if (t.consequence == \"LOF\") 1 \n",
+    "                        else if (t.consequence == \"MIS\") 2 \n",
+    "                        else 3)[0]''')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "### Learn more!\n",
```

Comment:
These links are broken when rendered on website. The actual website links are printed, and they are not clickable.